### PR TITLE
chore: Removes circular dependencies

### LINKED ISSUE #9337: [Feature Request] Removing circular dependencies in VBreadcrumbs, VTimePicker and VTreeview
### Problem to solve
To further improve the code quality, it would be nice to remove circular dependencies in `VBreadcrumbs`, `VTimePicker` and `VTreeview`.

In contrast to #7033, this is not a bug report. It won't become a bug as long as the top-level code avoids demanding something from its dependencies immediately. However, it is not limited to Rollup, but a general problem (which would also affect Webpack builds -- it is just not warning about this issue).

The circles are:

- `VBreadcrumbs\index.js` -> `VBreadcrumbs\VBreadcrumbs.js` -> `VBreadcrumbs\index.js`
- `VTimePicker\VTimePicker.js` -> `VTimePicker\VTimePickerTitle.js` -> `VTimePicker\VTimePicker.js`
- `VTreeview\VTreeviewNode.js` -> `VTreeview\VTreeviewNode.js`

### Proposed solution
# VBreadcrumbs

The reason for this circle seems to be `import { VBreadcrumbsDivider, VBreadcrumbsItem } from '.'` inside `VBreadcrumbs\VBreadcrumbs.js`.

While it could easily import `VBreadcrumbsItem` directly to avoid `'.'`, `VBreadcrumbsDivider` needs to be moved into its own file, to do the same:

`VBreadcrumbs\VBreadcrumbsDivider.js`
```diff
+ import { createSimpleFunctional } from '../../util/helpers'
+
+ export default createSimpleFunctional('v-breadcrumbs__divider', 'li')
```

`VBreadcrumbs\VBreadcrumbs.js`
```diff
- import { VBreadcrumbsDivider, VBreadcrumbsItem } from '.'
+ import VBreadcrumbsItem from './VBreadcrumbsItem'
+ import VBreadcrumbsDivider from './VBreadcrumbsDivider'
```

`VBreadcrumbs\index.js`
```diff
- import { createSimpleFunctional } from '../../util/helpers'
- 
- const VBreadcrumbsDivider = createSimpleFunctional('v-breadcrumbs__divider', 'li')
+ import VBreadcrumbsDivider from './VBreadcrumbsDivider'
```

# VTimePicker

The reason for this seems to be `SelectingTimes`. The easiest way is to would be to mode it out of the circle into its own file:

`VTimePicker\SelectingTimes.js`
```diff
+ enum SelectingTimes {
+   Hour = 1,
+   Minute = 2,
+   Second = 3
+ }
+
+ export { SelectingTimes }
```

`VTimePicker\VTimePickerTitle.js`
```diff
- import { SelectingTimes } from './VTimePicker'
+ import { SelectingTimes } from './SelectingTimes'
```

`VTimePicker\VTimePicker.js`
```diff
+ import { SelectingTimes } from './SelectingTimes'

...

- enum SelectingTimes {
-   Hour = 1,
-   Minute = 2,
-   Second = 3
- }
```

# VTreeviewNode

The self-dependency in `VTreeviewNode.js` occurs within a function call, which is not immediately invoked. Instead of the import, the circle could be shorted to:

```diff
- import VTreeviewNode from './VTreeviewNode'

...

- export default baseMixins.extend<options>().extend({
+ const component = baseMixins.extend<options>().extend({

...

    genChild (item: any): VNode {
-      return this.$createElement(VTreeviewNode, {
+      return this.$createElement(component, {

...

+ export default component; 
```

This won't fully resolve the this issue, but limits its scope to the minimum.

### PR DESCRIPTION
**Pull request by SebastianNiemann**
## Description
Removes circular dependencies in VBreadcrumbs, VTimePicker and VTreeview by either moving the critical dependency out of the circle (VBreadcrumbs and VTimePicker), or minimising its scope in case of a self-reference (VTreeview).

## Motivation and Context
Improves the code quality by resolving circular dependencies. Having these could create nasty bugs in the future. It also resolves warnings/errors from build tools and linters.
Closes #9337 and #7033.

## How Has This Been Tested?
Checking (by using Rollup) that all to be fixed circles where found before, but none after applying the changes.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:
- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)
- [ ] I've added new examples to the kitchen (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/vuetify/src/components/VTreeview/VTreeviewNode.ts [hunk 1]
```diff
@@ -86,7 +85,7 @@ export const VTreeviewNodeProps = {
 }
 
 /* @vue/component */
-export default baseMixins.extend<options>().extend({
+const component = baseMixins.extend<options>().extend({
```
Comment (KaelWD) [inline packages/vuetify/src/components/VTreeview/VTreeviewNode.ts]:
```suggestion
const VTreeviewNode = baseMixins.extend<options>().extend({
```

File: packages/vuetify/src/components/VTreeview/VTreeviewNode.ts [hunk 2]
```diff
@@ -270,7 +269,7 @@ export default baseMixins.extend<options>().extend({
       }), children)
     },
     genChild (item: any): VNode {
-      return this.$createElement(VTreeviewNode, {
+      return this.$createElement(component, {
```
Comment (KaelWD) [inline packages/vuetify/src/components/VTreeview/VTreeviewNode.ts]:
```suggestion
      return this.$createElement(VTreeviewNode, {
```

File: packages/vuetify/src/components/VTreeview/VTreeviewNode.ts [hunk 3]
```diff
@@ -334,3 +333,5 @@ export default baseMixins.extend<options>().extend({
     }, children)
   },
 })
+
+export default component
```
Comment (KaelWD) [inline packages/vuetify/src/components/VTreeview/VTreeviewNode.ts]:
```suggestion
export default VTreeviewNode
```

**Comment by SebastianNiemann** [comment]
@KaelWD Thx 👍  I applied all suggestions.

**Pull request merged**